internal/observer/observer.go

```diff
@@ -0,0 +1,13 @@
+package observer
+
+// Subject is an interface for objects that can be observed.
+type Subject[T VersionedConfig] interface {
```

Comment:
Not sure if I like the type constraint here and in `config_subject.go`. Alternatively, we could just use `any`. If we do stick with this, I should probably rename the package or the interfaces to include `VersionedConfig`.